File: src/main/java/com/dyonovan/neotech/common/container/storage/ContainerRFStorage.java
package com.dyonovan.neotech.common.container.storage;

import com.dyonovan.neotech.common.container.BaseContainer;
import com.dyonovan.neotech.common.tileentity.storage.TileBasicRFStorage;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraft.inventory.Slot;

public class ContainerRFStorage extends BaseContainer {

    public ContainerRFStorage(InventoryPlayer inventory, TileBasicRFStorage tile) {

        switch (tile.getSizeInventory()) {
            case 1:
                addSlotToContainer(new Slot(tile, TileBasicRFStorage.CHARGE_SLOT_1, 148, 18));
                break;
            case 2:
                addSlotToContainer(new Slot(tile, TileBasicRFStorage.CHARGE_SLOT_1, 148, 18));
                addSlotToContainer(new Slot(tile, TileBasicRFStorage.CHARGE_SLOT_2, 148, 36));
                break;
            case 3:
                addSlotToContainer(new Slot(tile, TileBasicRFStorage.CHARGE_SLOT_1, 148, 18));
                addSlotToContainer(new Slot(tile, TileBasicRFStorage.CHARGE_SLOT_2, 148, 36));
                addSlotToContainer(new Slot(tile, TileBasicRFStorage.CHARGE_SLOT_3, 148, 54));
                break;
        }
        bindPlayerInventory(inventory, 8, 84);
    }
}

File: src/main/scala/com/dyonovan/neotech/common/container/storage/ContainerRFStorage.scala
package com.dyonovan.neotech.common.container.storage

import com.dyonovan.neotech.common.tiles.storage.TileRFStorage
import com.teambr.bookshelf.common.container.BaseContainer
import net.minecraft.entity.player.InventoryPlayer

/**
  * This file was created for NeoTech
  *
  * NeoTech is licensed under the
  * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
  * http://creativecommons.org/licenses/by-nc-sa/4.0/
  *
  * @author Paul Davis <pauljoda>
  * @since 2/15/2016
  */
class ContainerRFStorage(playerInventory: InventoryPlayer, tile: TileRFStorage) extends
        BaseContainer(playerInventory, tile) {

    addSlotToContainer(new RestrictedSlot(tile, 0, 30, 20))
    addSlotToContainer(new RestrictedSlot(tile, 1, 30, 60))

    addPlayerInventorySlots(8, 84)
}
